command/server.go

```diff
@@ -1531,6 +1542,34 @@ func (c *ServerCommand) Run(args []string) int {
 	// Notify systemd that the server is ready (if applicable)
 	c.notifySystemd(systemd.SdNotifyReady)
 
+	if c.flagDev {
+		protocol := "http://"
+		if c.flagDevTLS {
+			protocol = "https://"
+		}
+		clusterJson.Nodes = []testcluster.ClusterNode{
+			{
+				APIAddress: protocol + config.Listeners[0].Address,
+			},
+		}
+		if c.flagDevTLS {
+			clusterJson.CACertPath = fmt.Sprintf("%s/%s", certDir, server.VaultDevCAFilename)
+		}
+	}
+
+	if c.flagDevClusterJson != "" {
```

Comment:
This seems redundant to me. The enableThreeNodeDevCluster function already has this check and writes out the file is the flag is set.

Reply:
Good point.  We still want to support this option for the single node (-dev) case, but I've changed it so that we don't do this twice in three-node mode (fa42073d6f993063395912c4fe9cbf6042e2cb9d).